packages/model/addon/-private/errors.js

```diff
@@ -427,17 +427,17 @@ export default ArrayProxy.extend(DeprecatedEvented, {
 
     ```app/routes/user/edit.js
     import Route from '@ember/routing/route';
+    import { action } from '@ember/object';
 
-    export default Route.extend({
-      actions: {
-        save: function(user) {
-          if (user.get('errors').has('email')) {
-            return alert('Please update your email before attempting to save.');
-          }
-          user.save();
+    export default class UserEditRoute extends Route {
+      @action
+      save(user) {
+        if (user.get('errors').has('email')) {
+          return alert('Please update your email before attempting to save.');
         }
+        user.save();
       }
-    });
+    }
```

Comment:
Should we convert this to a controller as well?

Reply:
yes - converted to controller